app/auth/apikey_crud.py

```diff
@@ -240,12 +246,11 @@ def check_key(self, api_key: str) -> dict | None:
                 )
                 conn.commit()
 
-                response["is_active"] = kc_info.is_enabled
-                response["iam_roles"] = kc_info.roles
+                # If user is not enabled in keycloak anymore
+                if not kc_info.is_enabled:
```

Comment:
To benefit from the DB cache, you need to update the information that the user is disabled in keycloak somewhere. See previous comment.